x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -221,7 +222,12 @@ public CacheFile getCacheFile(CacheKey cacheKey, long fileLength) throws Excepti
     @Override
     public IndexInput openInput(final String name, final IOContext context) throws IOException {
         ensureOpen();
+
         final BlobStoreIndexShardSnapshot.FileInfo fileInfo = fileInfo(name);
+        if (fileInfo.metadata().hashEqualsContents()) {
+            return new ByteArrayIndexInput("ByteArrayIndexInput(" + name + ')', fileInfo.metadata().hash().bytes);
```

Comment:
It looks like `fileInfo.metadata().hash()` is not today a slice of the underlying byte array (i.e. it has zero `offset` and full length) but I don't think we guarantee that in future. I think we should  pass offset and length to the `ByteArrayIndexInput` constructor just to be sure.

Reply:
Sure